DataConnectors/SentinelOne/SentinelOneSentinelConnector/__init__.py

```diff
@@ -49,14 +57,43 @@ def generate_date(self):
             logging.info("There is no last time point, trying to get events for last day.")
             past_time = (current_time - datetime.timedelta(days=1)).strftime("%Y-%m-%dT%H:%M:%S.%fZ")
         state.post(current_time.strftime("%Y-%m-%dT%H:%M:%S.%fZ"))
+        logging.info("Getting data from: " + past_time)
+        logging.info("To: " + current_time.strftime("%Y-%m-%dT%H:%M:%S.%fZ"))
         return (past_time, current_time.strftime("%Y-%m-%dT%H:%M:%S.%fZ"))
     
     def get_report(self, report_type_suffix, report_type_name, params = None):
         try:
             r = requests.get(self.domain + report_type_suffix, headers = self.header, params = params)
             if r.status_code == 200:
+                if("Threats" in report_type_name):
+                    self.get_threat_id(r.text)
+                
                 self.results_array_join(r.json(), report_type_name)
                 next_page_token = (r.json().get('pagination')).get('nextCursor')
+                logging.info("Report returns: {}".format(next_page_token))
+                return next_page_token
+            elif r.status_code == 400:
+                logging.error("Invalid user input received. See error details for further information."
+                      " Error code: {}".format(r.status_code))
+            elif r.status_code == 401:
+                logging.error("Unauthorized access - please sign in and retry. Error code: {}".format(r.status_code))
+            else:
+                logging.error("Something wrong. Error code: {}".format(r.status_code))
+        except Exception as err:
+            logging.error("Something wrong. Exception error text: {}".format(err))
+
+    def get_threat_data(self, report_type_suffix, report_type_name, id, params = None):
+        try:
+            if("Notes" in report_type_name):
+                url = self.domain + report_type_suffix + str(id) + "/notes"
+                r = requests.get(url, headers = self.header, params = params)
+            elif("Events" in report_type_name):
+                url = self.domain + report_type_suffix + str(id) + "/explore/events"
+                r = requests.get(url, headers = self.header, params = params)
+            if r.status_code == 200:
+                self.results_array_join(r.json(), report_type_name, id)
+                next_page_token = (r.json().get('pagination')).get('nextCursor')

```

Comment:
make sure in case there is no pagination object it should not throw exception.
and return None in case of `pagination` or `nextCursor` is not there in response

Reply:
I have checked on SentinelOne portal and it seems that pagination and nextCursor will always be present(it has null value when no next page, on that it does not throw an exception) except when there is some error in the request, on which it should throw exception.

Reply:
You just need to add a default value in the 2nd argument so that it will not through exception if there is any error on sentinel one side.